analytics/pubstack/pubstack_module_test.go

```diff
@@ -88,94 +89,100 @@ func TestPubstackModuleSuccess(t *testing.T) {
+		configTask := fakeConfigUpdateTask{}
+		module, err := NewModuleWithConfigTask(client, "scope", server.URL, 100, "1B", "1s", &configTask, clockMock)
+		assert.NoError(t, err, tt.description)
 
 		pubstack, _ := module.(*PubstackModule)
-		// attempt to log but no event channel was created because the feature is disabled in the original config
-		tt.logObject(pubstack)
-
-		// verify no event was received over a 10ms period
-		assertChanNone(t, intakeChannel, tt.description)
-
-		// allow time for the server to start serving the updated config
-		time.Sleep(10 * time.Millisecond)
-
-		// attempt to log; the event channel should have been created because the feature is enabled in updated config
-		tt.logObject(pubstack)
 
-		// verify an event was received within 10ms
-		assertChanOne(t, intakeChannel, tt.description)
+		// original config
+		configTask.Push(origConfig)
+		time.Sleep(10 * time.Millisecond)                            // allow time for the module to load the original config
+		tt.logObject(pubstack)                                       // attempt to log; no event channel created because feature is disabled in original config
+		clockMock.Add(1 * time.Second)                               // trigger event channel sending
+		assertChanNone(t, intakeChannel, tt.description+":original") // verify no event was received
+
+		// updated config
+		configTask.Push(updatedConfig)
+		time.Sleep(10 * time.Millisecond)                          // allow time for the server to start serving the updated config
+		tt.logObject(pubstack)                                     // attempt to log; event channel should be created because feature is enabled in updated config
+		clockMock.Add(1 * time.Second)                             // trigger event channel sending
+		assertChanOne(t, intakeChannel, tt.description+":updated") // verify an event was received
+
+		// no config change
+		configTask.Push(updatedConfig)
+		time.Sleep(10 * time.Millisecond)                            // allow time for the server to determine no config change
+		tt.logObject(pubstack)                                       // attempt to log; event channel should still be created from loading updated config
+		clockMock.Add(1 * time.Second)                               // trigger event channel sending
+		assertChanOne(t, intakeChannel, tt.description+":no_change") // verify an event was received
+
+		// shutdown
+		pubstack.sigTermCh <- os.Kill                                // simulate os shutdown signal
+		time.Sleep(10 * time.Millisecond)                            // allow time for the server to switch to shutdown generated config
+		tt.logObject(pubstack)                                       // attempt to log; event channel should be closed from the os kill signal
+		clockMock.Add(1 * time.Second)                               // trigger event channel sending
+		assertChanNone(t, intakeChannel, tt.description+":shutdown") // verify no event was received
```

Comment:
I'm wondering in this section, if the comments could be cutdown or refactored? It just seems like a lot of comments for one section, but I won't push back if you think they're all necessary.

Reply:
I thought it would help explain the flow of the test. It's otherwise not obvious. I kept the comments as one liners to allow for nice formatting. In production code I would extract each action to a well named function, but this is just a test so I thought comments are acceptable.